lib/model/query/audits.js

```diff
@@ -110,5 +110,56 @@ const getBySubmissionId = (submissionId, options) => ({ all }) =>
   _getBySubmissionId(all, options, submissionId);
 
 
-module.exports = { log, getLatestByAction, get, getBySubmissionId };
+
+const _getByEntityId = (fields, options, entityId) => sql`
+SELECT ${fields} FROM audits
+  LEFT JOIN actors ON actors.id=audits."actorId"
+  LEFT JOIN (
+    SELECT submissions.*, submission_defs."userAgent" FROM submissions
+    JOIN submission_defs ON submissions.id = submission_defs."submissionId" AND root -- just for the sake of userAgent which we don't need, we should remove it
+  ) submissions ON (audits.details->'submissionId')::INTEGER = submissions.id AND action IN ('entity.create', 'entity.update.version')
+  LEFT JOIN actors submission_actor ON submission_actor.id = submissions."submitterId"
+  LEFT JOIN submission_defs on submissions.id = submission_defs."submissionId" and submission_defs.current
+  LEFT JOIN actors current_submission_actor on current_submission_actor.id=submission_defs."submitterId"
+  LEFT JOIN audits approval ON (approval.details->'submissionId')::INTEGER = submissions.id AND approval.action = 'submission.update' AND (approval.details->>'reviewState')::TEXT = 'approved'
```

Comment:
Unless I'm misunderstanding, I think we need to handle the possibility of multiple approvals of the same submission. I've been wondering about that case, and I'm not sure how to select the correct approval when there are more than one. Maybe we could get away with selecting the submission's first approval? I could imagine edge cases where that doesn't work, though they seems pretty unlikely. For example:

- A submission is created for a form that can create entities, but the submission doesn't specify an entity. (That's possible, right?)
- The submission is approved. No entity is created.
- Someone later realizes that there was a mistake and edits the submission. This time, the submission version does specify an entity.
- The submission is reapproved, creating the entity.

If we want to be more robust about cases like that, maybe we select the first approval whose `processed` timestamp is greater than or equal to the `loggedAt` timestamp of the `entity.create` event ? Or maybe we add a column to the entity or entity source that identifies the specific approval that created the entity?

I think the most important thing though is to handle multiple approvals in some way. I think multiple approvals are a real possibility and that we'll want to avoid returning duplicate audit log entries in that case.